lib/_http_outgoing.js

```diff
@@ -430,6 +430,13 @@ function _storeHeader(firstLine, headers) {
   // wait until the first body chunk, or close(), is sent to flush,
   // UNLESS we're sending Expect: 100-continue.
   if (state.expect) this._send('');
+
+  // Test non-chunked message does not have trailer header set,
+  // message will be terminated by the first empty line after the
+  // header fields, regardless of the header fields present in the
+  // message, and thus cannot contain a message body or 'trailers'.
+  if (this.chunkedEncoding !== true && state.trailer)
+    throw new Error('Trailer header is invalid with this transfer encoding');
```

Comment:
It would probably be better to throw *before* we set `this._header`, otherwise the flush above could still send it.